plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseCaseInsensitiveMappingTest.java

```diff
@@ -172,13 +196,26 @@ public void testTableNameClash()
                     assertThat(computeActual("SHOW TABLES").getOnlyColumn().filter("casesensitivename"::equals)).hasSize(1); // TODO, should be 2
                     assertQueryFails("SHOW COLUMNS FROM casesensitivename", "Failed to find remote table name: Ambiguous name: casesensitivename");
                     assertQueryFails("SELECT * FROM casesensitivename", "Failed to find remote table name: Ambiguous name: casesensitivename");
+                    assertQueryFails("INSERT INTO casesensitivename VALUES 'a'", "Failed to find remote table name: Ambiguous name: casesensitivename");
+                    assertQueryFails("CREATE TABLE casesensitivename AS SELECT 'a' c", "Failed to find remote table name: Ambiguous name: casesensitivename");
                     assertQuery("SHOW COLUMNS FROM some_table", "SELECT 'd', 'varchar(5)', '', ''");
                     assertQueryReturnsEmptyResult("SELECT * FROM some_table");
                 }
             }
         }
     }
 
+    @Test
+    public void testCreateTableAsSelectClash()
+            throws Exception
+    {
+        try (AutoCloseable ignore1 = withTable("CaseSensitiveName", "(c varchar(5))")) {
+            assertThat(computeActual("SHOW TABLES").getOnlyColumn().filter("casesensitivename"::equals)).hasSize(1);
+            assertQueryFails("CREATE TABLE casesensitivename (c varchar(5))", ".*Table '.*' already exists");
```

Comment:
The test name is test**CreateTableAsSelect**Clash, but it's `CREATE TABLE`. How about creating a dedicated test method?